fix: correctly process multi-line quoted-printable headers

anton7c3 opened the pull request.

anton7c3 committed:
fix: correctly process multi-line quoted-printable headers

anton7c3 committed:
Merge pull request #1 from NeuraLegion/fix/multi-line-quoted-printable

fix: correctly process multi-line quoted-printable headers

anton7c3 committed:
fix: make sure non-encoded text stays in the headers

anton7c3 committed:
Merge pull request #2 from NeuraLegion/fix/quoted-printable-2

fix: make sure non-encoded text stays in the headers

anton7c3 committed:
feat: add the ability to parse base64 emails with no multipart

Add spec that checks base64 with multipart

anton7c3 committed:
Merge pull request #3 from NeuraLegion/feat/base64-emails

feat: add the ability to parse base64 emails with no multipart

anton7c3 commented:
@usiegl00 can we push this PR somehow please?

mverzilli commented on spec/crystal-mime_spec.cr:
```diff
@@ -4,15 +4,15 @@ describe MIME do
   it "Ensure test mail is RFC 1341 compliant" do
     # Ensure CRLF's are present in test:
-    f = File.read("spec/test-mime1.email")
+    f = {{ read_file("#{__DIR__}/test-mime1.email") }}
```
Would you share why you're changing from `File.read` to `read_file` in this PR?

anton7c3 replied:
`file.read` needs to have this file available in runtime by the path mentioned.
`{{ read_file }}` embeds the file contents into the source in compile time, which feels a little more robust.
In our project we have a ton of specs so we don't make `crystal spec <foo>`. We build a binary that has all the specs and then run it in parallel CI jobs with different examples started (Equivalent to `crystal spec -e`, but you build only once). So I just used the approach I'm kinda used to :)

mverzilli reviewed (approved):
Looks good! I'm just curious about the intention behind the change from `File.read` to `read_file`. A priori it seems it will shave off some test run time which will instead be spent as compile time?

YarikMamykin committed:
ci(workflows): add pr checks

YarikMamykin committed:
fix(all): lint

YarikMamykin committed:
ci(workflows): add specs job

YarikMamykin committed:
fix(mime): parsing of multipart inside multipart (#4)

* fix(mime): separate headers parsing into method

* fix(mime): implement parsing of inner multipart

* fix(mime): parsing of email with inner multiparts

* fix(mime): add spec for multipart inside multipart

* fix(mime): fix code style

* fix(mime): make boundary string construct more robust

* chore(mime): remove extra mime io

YarikMamykin committed:
ci(workflows): change base image

YarikMamykin committed:
fix(mime): format

anton7c3 committed:
ci(workflows): add pr checks

anton7c3 committed:
fix: correctly process quoted-printable bodies

Because of the bug in underlying library quoted-printable body crashes
the parser, because the library only expects "=\r\n" and not "=\n".

In addition fixed a crash when the header name and the header value are
placed on different lines

anton7c3 committed:
fix: correctly process quoted-printable bodies

Because of the bug in underlying library quoted-printable body crashes the parser, because the library only expects `=\r\n` and not `=\n`.

In addition fixed a crash when the header name and the header value are placed on different lines

anton7c3 committed:
fix: ensure both \n and \r\n work with quoted-printable

anton7c3 committed:
Merge pull request #7 from NeuraLegion/fix/even-more-line-endings

fix: ensure both \n and \r\n work with quoted-printable
